Feature 6  enemy

Abhishek981111 opened the pull request:
Enemy patrol movement on a platform

Abhishek981111 committed:
Added Idle, Run, Jump animations to player Prefab

Added Idle, Run, Jump animations to player prefab with transitions

Abhishek981111 committed:
Added Idle Jump and run animations

Added Idle, Jump and Run animations with transitions

Abhishek981111 committed:
Added transition from jump to run

Added transition from jump to run and added the conditions for animation

Abhishek981111 committed:
Update Player controller speed condition

Updated the player controller speed condition from 0.25 to 0.5 to transition between the animations.

Abhishek981111 committed:
Added crouch animation

Added crouch animation with transitions and conditions

Abhishek981111 committed:
test push only

Abhishek981111 committed:
commit for controller

Abhishek981111 committed:
Added Player Controller script and rigid body to player

Added Player controller script, rigidbody 2D, box collider 2D to player prefab. Also created a platform and added a box collider and sprite renderer

Abhishek981111 committed:
Scripting Animation & Flipping

Accept horizontal input from the user, play run animation and flip player according to the horizontal input.

Abhishek981111 committed:
added crouch and jump animation with animator script

added crouch and jump animation with animator script for playing respective animations

Abhishek981111 committed:
Player movement, jump, crouch

Added player movement logic to move in horizontal, with crouch and jump.

Abhishek981111 committed:
Added Movement logic for jump, crouch

Added movement for jump in vertical direction using AddForce method. Crouch animation changes.

Abhishek981111 committed:
Changes in jump animator window, script

Changes in jump animator, script for uneven jumps in heights. Fixed the issue.

Abhishek981111 committed:
Small change in collision naming convention

Abhishek981111 committed:
Tile Map Setup

Created ground tile map setup

Abhishek981111 committed:
Level Complete gameobject with LevelController Script

Created level complete gameobject with LevelController Script to load the next sample scene.

Abhishek981111 committed:
Player fall from platform and respawn

Added changes in PlayerController for player falling from platform and respawn

Abhishek981111 committed:
player fall from platform(Minor Changes) and respawn

Abhishek981111 committed:
Created KeyController, Key, score increment logic

Added Key game object, created a script for KeyController also added a score increment logic using ScoreController script.

Abhishek981111 committed:
Added simple rotation animation to key prefab

Abhishek981111 committed:
Added key going Up-Fade out animation

Abhishek981111 committed:
first enemy to the game, which patrols on the platforms.

Added EnemyController script for creating movement for enemy to patrol on platform.

Abhishek981111 committed:
Scene reload when touches with enemy, death animation

Abhishek981111 committed:
Added health system for player

Player health system is implemented

Abhishek981111 committed:
Scene reload after player health to zero

Abhishek981111 committed:
Added another enemy to the game and turned off the collision matrix between enemies

Abhishek981111 committed:
Changes in enemy movement script

Abhishek981111 committed:
Enemy movement and player health system

Revised code for enemy movement and player health system with UI.

Abhishek981111 committed:
Scene reloading on player death (Revised)

Reload scene when player health is zero.

SahayDarshan commented on Assets/Scripts/DeathCollider.cs:
```diff
@@ -0,0 +1,15 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
```
No need to create a separate script to check whether player collided with DeathCollider or not. You can implement this thing in your player controller and check the same thing using GetComponent or Tag. Always avoid MonoBehaviour classes and code repetition.

SahayDarshan commented on Assets/Scripts/EnemyController.cs:
```diff
@@ -0,0 +1,42 @@
+
+    private void PatrolEnemy()
+    {
+        enemyAnimator.SetBool("IsPatrol", true);
```
this particular line needs to be executed only once rather than every frame. You want to set your enemy animation to move , so you can do this in start or awake. No need to set to true again again when it is already true.

SahayDarshan commented on Assets/Scripts/EnemyController.cs:
```diff
@@ -0,0 +1,42 @@
+
+    private void OnCollisionEnter2D(Collision2D collision)
+    {
+        if(collision.transform.GetComponent<PlayerController>() != null)
```
This can be simplified in a much better way.
`if(collision.transform.TryGetComponent<PlayerController>(PlayerController controller))
{
    controller.DecreaseHealth();
}`

Now what actually is happening here is TryGetComponent works same as GetComponent but it also helps in storing the reference of that particular component in a variable which you can use later.
For example in this case i have controller variable which is of type PlayerController. No need to repeat same code again and again.

SahayDarshan commented on Assets/Scripts/LevelOverController.cs:
```diff
@@ -0,0 +1,14 @@
+public class LevelOverController : MonoBehaviour
+{
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
```
again repeatition of logic which can be implemented in PlayerController. No need to create separate scripts for such small uses.

SahayDarshan commented on Assets/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,190 @@
+        {
+            isCrouching = true;
+            animator.SetBool("Crouch", isCrouching);
+            boxCol.size = new Vector3(colliderSizeInX, colliderSizeInY);
```
Avoid Setting the size of colliders through scripts. Use animation window to adjust the size of collider using keyframes.

SahayDarshan reviewed (commented):
I have mentioned few changes that you should do to make a more clean and optimized code.

Abhishek981111 committed:
Code optimization for death collider and crouching

death collider controlled through player controller and crouching through animation window using keyframes.

Abhishek981111 committed:
Code optimization in EnemyController script

Abhishek981111 committed:
Fix for death animation

Created a coroutine for playing death animation before reloading scene.
